src/app.jsx

```diff
@@ -1,11 +1,49 @@
 import React from 'react';
+import {
+  BrowserRouter as Router, Route, Link, Switch,
+} from 'react-router-dom';
 import './index.css';
 
-const App = () => (
+// Temporary functional components will be deleted after project setup.
+const Home = () => (
+  <div className="container">
+    <h1 className="app-title">Home</h1>
+    <p>
+      <Link to="/sign-up">Sign up Now </Link>
+    </p>
+  </div>
+);
+
+const About = () => (
+  <div className="container">
+    <h1 className="app-title" style={{ color: '#099EBF' }}>
+      About
+    </h1>
+    <p>
+      <Link to="/">Go back home </Link>
+    </p>
+  </div>
+);
+
+const SignUp = () => (
   <div className="container">
-    <h1 className="app-title">Authors Haven</h1>
-    <p>Welcome!</p>
+    <h1 className="app-title" style={{ color: '#09BF86' }}>
+      Sign up
+    </h1>
+    <p>
+      <Link to="/about">About </Link>
+    </p>
   </div>
 );
 
+const App = () => (
+  <Router>
+    <Switch>
+      <Route exact path="/" component={Home} />
+      <Route path="/about" component={About} />
```

Comment:
Although, they may be removed later. Let's have it serve an exact path.

Reply:
@me-x-mi I did not get this comment well, please feed me with some more insights

Reply:
Let me send an example for vivid description.
```javascript 
<Route exact path="/about" component={About} />
```

Reply:
@me-x-mi if you use the <Switch> you do not need to put `exact` on each route because your like repeating yourself if am not mistaken here

Reply:
What about if I have a URL like about/readmore 
```javascript 
<Route exact path="/about/readme" component={About} />
```
Will this not serve /about?

Reply:
This is true, It will serve the first match route which is /about. he has to put to add `exact`  on the route that can have a descendant like put exact on `/users` when you have `/user/<something else>

Reply:
@me-x-mi @aaronsekisambu  `Exact should be added on the parent route`. To mean that: 
```javascript
<Route exact path="/about/readme" component={About} /> \\ will serve /about
``` 
To avoid that we have to add `exact` on the parent route even when we are using `switch`
```javascript
<Route exact path="/about" component={About} />
<Route path="/about/readme" component={About} />    \\ this will serve the correct route
```